pkg/checker.go

```diff
@@ -0,0 +1,155 @@
+package pkg
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+)
+
+// URL information
+type URL struct {
+	ip     string
+	port   int
+	host   string
+	scheme string
+	path   string
+}
+
+// CheckConfig : Check every record of DNS name agaist config information
+// Returns check result and metrics
+func CheckConfig(args map[string]interface{}) []*CheckResult {
```

Comment:
this method should probably be moved to a`http` package and renamed `Check(check HttpCheck) []*CheckResult`